src/hooks/project-hooks.tsx

```diff
@@ -108,14 +138,24 @@ export const ProjectProvider = ({
       type: "edit-in-makecode",
     });
     await doAfterEditorUpdate(() => {
-      setEditorOpen(true);
+      navigate(createCodePageUrl());
       return Promise.resolve();
     });
-  }, [doAfterEditorUpdate, logging, setEditorOpen]);
-
+  }, [doAfterEditorUpdate, logging, navigate]);
+  const browserNavigationToEditor = useCallback(async () => {
+    try {
+      await doAfterEditorUpdate(() => {
+        return Promise.resolve();
+      });
+      return true;
+    } catch (e) {
+      // doAfterEditorUpdate only fails when navigating to the code page directly.
+      // In this case, the caller of browserNavigationToEditor redirects.
+      return false;
```

Comment:
Maybe roughly this? I think most of the value is in having CodeEditorError in the two places to make the flow clearer without relying on comments.

```
if (e instanceof CodeEditorError) {
  // Comment as above
  return false;
}
// Unexpected error, can't handle better than the redirect
logging.error(e)
return false;
```

Reply:
[a8ab3da](https://github.com/microbit-foundation/ml-trainer/pull/416/commits/a8ab3dad585aa59ebea6831537eacd17dd76a9db)